compiler/semtypes.nim

```diff
@@ -1990,8 +1990,8 @@ proc semTypeNode(c: PContext, n: PNode, prev: PType): PType =
       result = semProcTypeWithScope(c, n, prev, skProc)
   of nkEnumTy: result = semEnum(c, n, prev)
   of nkType: result = n.typ
-  of nkStmtListType: result = semStmtListType(c, n, prev)
-  of nkBlockType: result = semBlockType(c, n, prev)
+  of nkStmtListType, nkStmtList: result = semStmtListType(c, n, prev)
+  of nkBlockType, nkBlockStmt, nkBlockExpr: result = semBlockType(c, n, prev)
   else:
     localError(c.config, n.info, "type expected, but got: " & renderTree(n))
```

Comment:
I'm not sure if this part is needed for the PR anymore, but I would like to extend these changes and just get rid of the error here and call `semTypeExpr` which gives the exact same error on non-types, but I don't know why `semTypeNode` is like this already. I know that you can make any expression work here by calling an identity template, so it doesn't make any sense to me. Might warrant an RFC but I don't know how to describe what happens here.